cmd/ctlog/verifyfulcio/main.go

```diff
@@ -0,0 +1,136 @@
+	flag.Parse()
+	var strictMatch = flag.Bool("strict", true, "If set to true ctlog must only contain the Fulcios in the list, no more, no less")
+	ctx := signals.NewContext()
+	fulcioURLs := make([]*url.URL, 0, len(fulcioList))
+	for _, f := range fulcioList {
+		u, err := url.Parse(f)
+		if err != nil {
+			logging.FromContext(ctx).Fatalf("Invalid fulcioURL %s : %v", f, err)
+		}
+		fulcioURLs = append(fulcioURLs, u)
+	}
+	fmt.Printf("GOT: %s\n", fulcioList.String())
+
+	// First grab the certs that CTLog has.
+	ctlog := fmt.Sprintf("%s/%s/ct/v1/get-roots", *ctlogURL, *ctlogPrefix)
+	ctlogResponse, err := http.Get(ctlog)
+	if err != nil {
+		logging.FromContext(ctx).Fatalf("Failed to get trusted certs from ctlog: %v", err)
+	}
+	defer ctlogResponse.Body.Close()
+	body, err := io.ReadAll(ctlogResponse.Body)
+	if err != nil {
+		logging.FromContext(ctx).Fatalf("Failed to read body from ctlog: %v", err)
+	}
+	certs := CertResponse{}
+	if err := json.Unmarshal(body, &certs); err != nil {
+		logging.FromContext(ctx).Fatalf("Failed to unmarshal body from ctlog: %v", err)
+	}
+	for i, cert := range certs.Certificates {
+		logging.FromContext(ctx).Infof("Got back cert %d: %s", i, cert)
+	}
+
+	// Keep track of certs found. Same index as fulcioURLs
+	certsFound := make([]bool, len(fulcioURLs))
+	for i, fulcio := range fulcioURLs {
+		logging.FromContext(ctx).Infof("Fetching fulcio cert %s", fulcio)
+		client := fulcioclient.NewClient(fulcio)
+		root, err := client.RootCert()
+		if err != nil {
+			logging.FromContext(ctx).Fatalf("Failed to fetch fulcio Root %s cert: %v", fulcio.String(), err)
+		}
+		fulcioCerts, err := cryptoutils.UnmarshalCertificatesFromPEM(root.ChainPEM)
+		if err != nil {
+			logging.FromContext(ctx).Fatalf("Failed to unmarshal fulcio Root cert %s cert: %v", fulcio.String(), err)
+		}
+		fulcioRoot, err := cryptoutils.MarshalCertificateToPEM(fulcioCerts[len(fulcioCerts)-1])
+		if err != nil {
+			logging.FromContext(ctx).Fatalf("Failed to marshal fulcio Root cert %s cert: %w", fulcio.String(), err)
+		}
+		logging.FromContext(ctx).Infof("Got a root cert for fulcio Root cert %s", fulcio)
+		// Strip the certificate begin/end marker strings since CTLog doesn't
+		// have those.
+		block, _ := pem.Decode(fulcioRoot)
+		if err != nil {
+			logging.FromContext(ctx).Fatalf("Failed to decode fulcio Root PEM %s cert: %w", fulcio.String(), err)
+		}
+		fulcioRootPEM := []byte(base64.StdEncoding.EncodeToString(block.Bytes))
+		for j := range certs.Certificates {
+			logging.FromContext(ctx).Infof("Checking ctlog root cert %s", certs.Certificates[j])
+			if bytes.Compare(fulcioRootPEM, []byte(certs.Certificates[j])) == 0 {
+				logging.FromContext(ctx).Infof("Found a matching root cert for fulcio Root cert %s cert: %w", fulcio.String(), err)
+				certsFound[i] = true
+			}
+		}
+	}
+	// Check that all the expected roots were found.
```

Comment:
As mentioned above, if we're supporting third-party CAs, we would want to check that we've found all fulcio roots, but it might just be a subset of roots in the ctlog config.

Reply:
Thought we might need that, so thanks for that use case :) 

There's the --strict flag that we can then flip to false. For the tests, I wanted to make sure it was a precise match, but thought there would be other use cases.
https://github.com/sigstore/scaffolding/pull/352/files#diff-f3a6ec0f5f1577f486baea8af21fd6f52dc4a009ed193babaef0b758e09eb841R57

Reply:
Looks good!